### PR TITLE
Add release notes for 2.0.1

### PR DESCRIPTION
**Pull request by gassmoeller**
Added release notes for 2.0.1 with descriptions for all the fixed bugs.

### CODE REVIEW

File: doc/release-tasklist [hunk 1]
```diff
@@ -96,6 +96,45 @@
...
+  inconsistently with its description. Its first entry did not correspond
+  to the background field, but to the first compositional field, and the
+  last value was ignored. This is fixed now, the first entry is used for
+  the background field, and all following values determine whether to include
+  the corresponding compositional fields.
+- Fixed: The 'depth dependent' material model did not properly initialize
+  the material model it uses as a base model. This caused crashes if the
+  base model requires an initialization (such as the 'steinberger' material
+  model). This is fixed now by properly initializing the base model.
+- Fixed: The advection assembler for DG elements was not thread-safe when,
```
Comment (tjhei) [inline doc/release-tasklist]: remove "when"

File: doc/release-tasklist [hunk 2]
```diff
@@ -96,6 +96,45 @@
...
+  which lead to wrong results or crashes if a discontinuous temperature
+  or composition discretization was combined with multithreading.
+  This is fixed now.
```
Comment (tjhei) [inline doc/release-tasklist]: I would remove this sentence.

Comment (gassmoeller) [reply doc/release-tasklist]: @jdannberg asked me to put this in. Do you feel strongly about removing it? I addressed the other comment.

**Comment by gassmoeller** [comment]
Addressed both comments.

File: doc/release-tasklist [hunk 3]
```diff
@@ -96,6 +96,44 @@
...
+  to the background field, but to the first compositional field, and the
+  last value was ignored. This is fixed now, the first entry is used for
+  the background field, and all following values determine whether to include
+  the corresponding compositional fields.
+- Fixed: The 'depth dependent' material model did not properly initialize
+  the material model it uses as a base model. This caused crashes if the
+  base model requires an initialization (such as the 'steinberger' material
+  model). This is fixed now by properly initializing the base model.
+- Fixed: The advection assembler for DG elements was not thread-safe,
+  which lead to wrong results or crashes if a discontinuous temperature
```
Comment (bangerth) [inline doc/release-tasklist]: "led" is, despite widespread use of "lead", the correct past tense of "lead":

http://www.write.com/writing-guides/general-writing/word-choice/lead-versus-led/

File: doc/release-tasklist [hunk 4]
```diff
@@ -96,6 +96,44 @@
...
+  or composition discretization was combined with multithreading.
+- Disabled: The particle functionality was not tested when combined with a
+  free surface boundary, and is currently not supported. This is now made
```
Comment (bangerth) [inline doc/release-tasklist]: is -> this combination

**Comment by gassmoeller** [comment]
I think I addressed your comments.

**Pull request merged**